api/src/main/java/org/openmrs/api/ConceptService.java

```diff
@@ -1922,5 +1922,57 @@ public Drug getDrugByMapping(String code, ConceptSource conceptSource,
 	 */
 	@Authorized(PrivilegeConstants.GET_CONCEPTS)
 	boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType);
-	
+	/**
+	 * Get Concept by id or name convenience method
+	 * 
+	 * @param idOrMap
+	 * @return the found Concept
+	 * @throws APIException
+	 */
+	@Authorized(PrivilegeConstants.GET_CONCEPTS)
+	public static Concept getConceptInfo(String idOrMap) throws APIException {
+
+		Concept cpt = null;
+
+		if (idOrMap != null) {
+
+			idOrMap = idOrMap.trim();
+
+			//if input contains at least one period handle recursively as a code constant
+			if (idOrMap.contains(".")) {
+				return getConcept(evaluateStaticConstant(idOrMap));
+			}
+
+			// handle  mapping id: xyz:ht
+			int index = id.indexOf(":");
+			if (index != -1) {
+				cpt = Context.getService(HtmlFormEntryService.class).getConceptByMapping(idOrMap);
+				if (cpt != null) {
+					return cpt;
+				}
+			}
+
+			// handle uuid id: "a3e1302b-74bf-11df-9768-17cfc9833272", if the id matches a uuid format
+			if (isValidUuidFormat(idOrMap)) {
+				cpt = Context.getConceptService().getConceptByUuid(idOrMap);
+				if (cpt != null) {
+					return cpt;
+				}
+			}
+
+			//finally, see if this is a parseable int; if so, try looking up concept by id
+			try { //handle integer: idOrMap
```

Comment:
To avoid using try catch to check if string is parsable you can make use of isParsable method in NumberUtils class.
[https://commons.apache.org/proper/commons-lang/javadocs/api-3.4/org/apache/commons/lang3/math/NumberUtils.html#isNumber(java.lang.String)](url)